### PR TITLE
NAS-119049 / 22.12.1 / Nas-119049 - Fixed LDAP slide-in is sometimes blank

### PR DESCRIPTION
**Pull request by Raz-Dva**
Go to Credentials ->  Configure Active Directory 

image

### CODE REVIEW

**Review by undsoft — CHANGES_REQUESTED** [review]
I don't really understand what these changes do or how they should help. Can you explain?

**Comment by Raz-Dva** [comment]
If click a row in Сertificate Signing Requests and create an ACME certificate(button), the slide is blank. 
The reason for this 
`  onCreateAcmePressed(): void {
    this.slideInService.close();
    const acmeForm = this.slideInService.open(CertificateAcmeAddComponent);
  }`
The code in setTimeout (`this.slideInBody.clear()`) will run after the component instance is created in the `open` method

File: src/app/modules/ix-forms/components/ix-slide-in/ix-slide-in.component.ts [hunk 1]
```diff
@@ -67,19 +69,24 @@ export class IxSlideInComponent implements OnInit, OnDestroy {
 
   closeSlideIn(): void {
     this.isSlideInOpen = false;
-
-    setTimeout(() => {
+    this.clearBodyExpected = true;
```
Comment (undsoft) [inline src/app/modules/ix-forms/components/ix-slide-in/ix-slide-in.component.ts:72]: Let's rename to `wasBodyCleared` instead.

File: src/app/modules/ix-forms/components/ix-slide-in/ix-slide-in.component.ts [hunk 2]
```diff
@@ -29,6 +29,8 @@ export class IxSlideInComponent implements OnInit, OnDestroy {
   isSlideInOpen = false;
   wide = false;
   private element: HTMLElement;
+  timeOutOfClear: any;
```
Comment (undsoft) [inline src/app/modules/ix-forms/components/ix-slide-in/ix-slide-in.component.ts:32]: There is a `Timeout` type in our project for this.

**Review by undsoft — APPROVED** [review]
I now see what you're trying to do. Make sense and good catch.
It's a good idea to also clear timeout if it's set in `ngOnDestroy`.

**Pull request merged**